ci_tools/check_slots.py

```diff
@@ -26,82 +28,132 @@
+                        inspect.getfile(mod), inspect.getsourcelines(cls_obj)[1], inspect.getsourcelines(cls_obj)[1], "failure", f"Missing attribute nodes in : `{mod_name}.{cls_name}`"))
 
         if all_attr and cls_name not in all_attr:
-            missing_from_all.append(f"{mod_name}.{cls_name}")
-
-print("Missing __all__")
-print(missing_all)
-print("__all__ non-alphabetical")
-print(non_alphabetical_all)
-print("Missing __slots__")
-print(missing_slots)
-print("Missing _attribute_nodes")
-print(missing_attribute_nodes)
-print("Not in __all__")
-print(missing_from_all)
-print("Misused slots")
-print(overridden_slots)
-
-with open(args.output, "w", encoding="utf-8") as out:
+            ErrorCollection['missing_from_all'].append(fill_dictionary("Classes missing from `__all__`", f"`pyccel.ast.{mod_name}.{cls_name}`",
+                inspect.getfile(mod), inspect.getsourcelines(cls_obj)[1], inspect.getsourcelines(cls_obj)[1], "failure", f"`{mod_name}.{cls_name}` classe is missing from `__all__`"))
+
+messages = extract_dict_elements(ErrorCollection)
+if not messages['annotations']:
+    messages['summary'] = "Check Slots\n\nSuccess:The operation was successfully completed. All necessary tasks have been executed without any errors or warnings.\n\n"
+    messages.pop('annotations')
+json_data = json.dumps(messages)
+with open('test_json_result.json', mode='w', encoding="utf-8") as json_file:
+    json_file.write(json_data)
+
+with open(args.output, "w", encoding="utf-8") as md_file:
     # Report error
-    if missing_all:
-        print("## Missing `__all__`", file=out)
-        for f in missing_all:
-            print(f"-   `pyccel.ast.{f}`", file=out)
-    if non_alphabetical_all:
-        print("## Non-alphabetical `__all__`", file=out)
-        for f in non_alphabetical_all:
-            print(f"-   `pyccel.ast.{f}`", file=out)
-    if missing_from_all:
-        print("## Classes missing from `__all__`", file=out)
-        for f in missing_from_all:
-            print(f"-   `pyccel.ast.{f}`", file=out)
-    if missing_slots:
-        print("## Classes with no `__slots__`", file=out)
-        for f in missing_slots:
-            print(f"-   `pyccel.ast.{f}`", file=out)
-    if missing_attribute_nodes:
-        print("## Classes with no `_attribute_nodes`", file=out)
-        for f in missing_attribute_nodes:
-            print(f"-   `pyccel.ast.{f}`", file=out)
-    if overridden_slots:
-        print("## Misused slots", file=out)
-        for o in overridden_slots:
-            print("-  ", o, file=out)
-
-failure = (bool(missing_all) or # bool(non_alphabetical_all) or
-          bool(missing_slots) or bool(missing_attribute_nodes) or
-          bool(overridden_slots))
+    md_file.write("# " + messages['title'] + '\n\n')
+    index = messages['summary'].find('\n\n')
+    md_file.write("## " + messages['summary'][:index])
```

Comment:
I think it would be safer to put the `##` inside the summary. That way you don't have to worry about whether or not it's the first header and it won't cause problems if we reorted the messages